t/t-pointer.sh

```diff
@@ -330,8 +330,8 @@ begin_test "pointer --check (with invalid pointer)"
 
   echo "not-a-pointer" > bad.ptr
 
-  ! git lfs pointer --check --file bad.ptr
-  ! git lfs pointer --check --stdin < bad.ptr
+  git lfs pointer --check --file bad.ptr && exit 1 || true
+  git lfs pointer --check --stdin < bad.ptr && exit 1 || true
```

Comment:
Looking again at this comment thread, I see there may also be another minor inconsistency with the `"pointer --check (with invalid arguments)"` test (which already exists, to be fair, and is not part of this PR).  It looks like:
```go
begin_test "pointer --check (with invalid arguments)"
(
  set -e

  reponame="pointer---check-invalid-pointer"
  git init "$reponame"
  cd "$reponame"

  touch a.txt

  # git-lfs-pointer(1) --check with invalid combination --compare
  ! git lfs pointer --check --compare

  # git-lfs-pointer(1) --check without --file or --stdin
  ! git lfs pointer --check

  # git-lfs-pointer(1) --check with --file and --stdin
  ! git lfs pointer --check --file a.txt --stdin
)
end_test
```

This might be a good time to change the `reponame` string as well as turn the `! git lfs pointer ...` commands into `git lfs pointer ... && exit 1`, e.g.:
```go
begin_test "pointer --check (with invalid arguments)"
(
  set -e

  reponame="pointer---check-invalid-arguments"
  git init "$reponame"
  cd "$reponame"

  touch a.txt

  # git-lfs-pointer(1) --check with invalid combination --compare
  git lfs pointer --check --compare && exit 1

  # git-lfs-pointer(1) --check without --file or --stdin
  git lfs pointer --check && exit  1

  # git-lfs-pointer(1) --check with --file and --stdin
  git lfs pointer --check --file a.txt --stdin && exit 1
)
end_test
```